woocommerce/woocommerce-android
# Upgrade Gradle to 7.4 & AGP to 7.1.1

This PR upgrades `Gradle to 7.4` & `AGP to 7.1.1`.

### Description


It includes the following:
- `Gradle` version upgraded to `7.3.3` with the `./gradlew wrapper --gradle-version=7.3.3 --distribution-type=all` command.
- `AGP` version upgrade to `7.0.4` (see `settings.gradle` change).
- Build output diff:
  - To identify and fix new warnings/errors that got added.
  - To identify and verify old warnings/errors that got removed.
- Lint output diff
  - To identify and fix new warnings/errors that got added.
  - To identify and verify old warnings/errors that got removed.
- The above 4 steps were redone with a follow-up update to `Gradle 7.4` & `AGP 7.1.1`.

-----

It updates the following libraries:
- [Navigation to 2.4.1.](https://github.com/woocommerce/woocommerce-android/pull/5755/commits/7071c72a8d5847608cd5e25fddc3286d0c81f4b2)
- [Compose to 1.1.0-rc01.](https://github.com/woocommerce/woocommerce-android/pull/5755/commits/07324e7125eda3c7d29a85b47cfa7445ce5cfb05)

It resolves the following test issues:
- [Remove unnecessary duplicate internal coroutines api annotation.](https://github.com/woocommerce/woocommerce-android/pull/5755/commits/0fa3c1fe3a64efc7e1ba74dacdcbb81a68c08ad3)

It created the following GitHub issues:
- [Gradle - Unable to Strip Some Libraries, Packaging as They Are #5761](https://github.com/woocommerce/woocommerce-android/issues/5761)

-----

Besides the above, you can reference the `Gradle 7.4 & AGP 7.1.1` branches of the below libraries for testing/verification purposes:
- [x] [WordPress-FluxC-Android](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2258)
- [x] [WordPress-Login-Flow-Android](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/83)
- [x] [WordPress-MediaPicker-Android](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/55)

**NOTE**: This is a draft PR because all `Gradle & AGP` upgrade PRs need to be merged together.

### Merging instructions


- [x] Update all 3 library references to PR hash (this is done for testing purposes):
  - [x] [fluxCVersion](https://github.com/woocommerce/woocommerce-android/pull/5755/commits/5fc15ab83c9a4fc53fa9f025ce927d68d9af65c4)
  - [x] [wordPressLoginVersion](https://github.com/woocommerce/woocommerce-android/pull/5755/commits/13e1c3028226160a486f53275173195dfdae681b)
  - [x] [mediapickerVersion](https://github.com/woocommerce/woocommerce-android/pull/5755/commits/71c3910b32b9fcb21ffbcd5a4d52011dc1924d02)
- [x] Mark PR as `Ready for review` and assign https://github.com/orgs/wordpress-mobile/teams/owl-team to it.
- [x] Merge all 3 libraries into `trunk` (after them being code reviewed and approved)
  - [x] [WordPress-FluxC-Android](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2258)
  - [x] [WordPress-Login-Flow-Android](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/83)
  - [x] [WordPress-MediaPicker-Android](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/55)
- [x] Update all 3 library references to `trunk` hash or tag:
  - [x] [fluxCVersion](https://github.com/woocommerce/woocommerce-android/pull/5755/commits/bfcaeecdb0d1023a50dfcbc389318334658702ff)
  - [x] [wordPressLoginVersion](https://github.com/woocommerce/woocommerce-android/pull/5755/commits/c4e0ddaa25566e85ab9a6d2cf13c432c43293b32)
  - [x] [mediapickerVersion](https://github.com/woocommerce/woocommerce-android/pull/5755/commits/05040c3d7e743929bce0de8d32c7fe5250021bc5)
- [x] Remove `status: do not merge` label.

### Testing instructions


- Smoke test the app.
- Use composite build for all 3 libraries (following the instructions in [local-builds.gradle-example](https://github.com/woocommerce/woocommerce-android/blob/develop/local-builds.gradle-example) file) and smoke test the app again. PS: Make sure you're on the `Gradle & AGP` branches on all 3 libraries.

-----
